### PR TITLE
Adding support for passphrase for private key files 

### PR DESCRIPTION
**Pull request by iam404**
Fix Issue #314

### CODE REVIEW

File: lib/kafka/client.rb [hunk 1]
```diff
@@ -436,15 +436,16 @@ def initialize_cluster
...
+    def build_ssl_context(ca_cert, client_cert, client_cert_key, client_cert_key_passphrase)
       return nil unless ca_cert || client_cert || client_cert_key
 
       ssl_context = OpenSSL::SSL::SSLContext.new
 
       if client_cert && client_cert_key
         ssl_context.set_params(
           cert: OpenSSL::X509::Certificate.new(client_cert),
-          key: OpenSSL::PKey.read(client_cert_key)
+          key: OpenSSL::PKey.read(client_cert_key, client_cert_key_passphrase)
```
Comment (tcrayford) [inline lib/kafka/client.rb]: what happens here if you run it past a client cert key that doesn't have a passphrase?

**Comment by tcrayford** [comment]
I'd like to see tests for this

File: lib/kafka/client.rb (hunk 1 above)
Comment (iam404) [inline lib/kafka/client.rb]: If `ssl_client_cert_key_passphrase` is not pass then default  value `nil` is saved. Also openssl tend to use passphrase only when key is encrypted. If it is not encrypted then passphrase provided is not used.
https://github.com/ruby/openssl/blob/7b319d95ed46a9ffb1d9962d5d4de8bff94cb28c/ext/openssl/ossl_pkey_rsa.c#L210

**Comment by dasch** [comment]
@iam404 do you expect to work more on this, or should I close it?

**Comment by dasch** [comment]
Closed due to lack of activity. @iam404 I can reopen if you'd like to continue to work on this.

**Pull request closed without merging**